x-pack/plugins/apm/server/lib/services/profiling/get_service_profiling_statistics.ts

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { keyBy, last } from 'lodash';
+import { Logger } from 'kibana/server';
+import util from 'util';
+import { maybe } from '../../../../common/utils/maybe';
+import { ProfileStackFrame } from '../../../../typings/es_schemas/ui/profile';
+import {
+  ProfilingValueType,
+  ProfileNode,
+  getValueTypeConfig,
+} from '../../../../common/profiling';
+import { ProcessorEvent } from '../../../../common/processor_event';
+import { ESFilter } from '../../../../../../typings/elasticsearch';
+import {
+  PROFILE_STACK,
+  PROFILE_TOP_ID,
+  SERVICE_NAME,
+} from '../../../../common/elasticsearch_fieldnames';
+import { rangeQuery, environmentQuery } from '../../../../common/utils/queries';
+import { APMEventClient } from '../../helpers/create_es_client/create_apm_event_client';
+import { Setup, SetupTimeRange } from '../../helpers/setup_request';
+import { withApmSpan } from '../../../utils/with_apm_span';
+
+const MAX_STACK_IDS = 10000;
+const MAX_STACKS_PER_REQUEST = 1000;
+
+const maybeAdd = (to: any[], value: any) => {
+  if (to.includes(value)) {
+    return;
+  }
+
+  to.push(value);
+};
+
+function getProfilingStats({
+  apmEventClient,
+  filter,
+  valueTypeField,
+}: {
+  apmEventClient: APMEventClient;
+  filter: ESFilter[];
+  valueTypeField: string;
+}) {
+  return withApmSpan('get_profile_stats', async () => {
```

Comment:
Why don't we name these the same as the function that's calling them?

Reply:
I've changed the name slightly. In general that's a good approach, but we have to keep in mind that these span names should be unique and preferably recognisable across all custom spans, so it's easier for us to understand them at a glance.